src/Plutarch/Extra/FixedDecimal.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Plutarch.Extra.FixedDecimal (
+    PFixedDecimal (..),
+    DivideSemigroup (..),
+    DivideMonoid (..),
+    decimalToAdaValue,
+) where
+
+import Plutarch.Api.V1
+import Plutarch.Api.V1.Value
+import Plutarch.Bool
+import Plutarch.Integer
+import Plutarch.Prelude
+import Plutarch.Unsafe
+
+decimalUnit :: Term s PInteger
+decimalUnit = 1000000
+
+{- | Fixed width decimal. Decimal point is at 1,000,000.
+ This would be used for representing Ada value with some Lovelace changes.
+
+ @since 1.0.0
+-}
+newtype PFixedDecimal (s :: S)
+    = PFixedDecimal (Term s PInteger)
+
+-- | @since 1.0.0
+deriving via
+    (DerivePNewtype PFixedDecimal PInteger)
+    instance
+        (PlutusType PFixedDecimal)
+
+-- | @since 1.0.0
+deriving via
+    (DerivePNewtype PFixedDecimal PInteger)
+    instance
+        PEq PFixedDecimal
+
+-- | @since 1.0.0
+deriving via
+    (DerivePNewtype PFixedDecimal PInteger)
+    instance
+        POrd PFixedDecimal
+
+-- | @since 1.0.0
+instance Num (Term s PFixedDecimal) where
+    (+) = (+)
+    (-) = (-)
+    (punsafeCoerce -> x) * (punsafeCoerce -> y) =
+        punsafeCoerce $ pdiv # (x * y) # decimalUnit
```

Comment:
should prefer more specific coercions (pto, pcon in this case I think) where they are available - it's slightly safer